src/server/pfs/server/api_server.go

```diff
@@ -71,23 +72,23 @@ func (a *apiServer) ActivateAuth(ctx context.Context, request *pfs.ActivateAuthR
 
 func (a *apiServer) ActivateAuthInTransaction(txnCtx *txncontext.TransactionContext, request *pfs.ActivateAuthRequest) (response *pfs.ActivateAuthResponse, retErr error) {
 	// Create role bindings for projects created before auth activation
-	var projectInfo pfs.ProjectInfo
-	if err := a.driver.projects.ReadWrite(txnCtx.SqlTx).List(&projectInfo, col.DefaultOptions(), func(string) error {
+	projIter, err := coredb.ListProject(txnCtx.Context(), a.env.DB)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list projects")
+	}
+	projInfo := &pfs.ProjectInfo{}
+	for !errors.Is(projIter.Next(txnCtx.Context(), projInfo), io.EOF) {
```

Comment:
This should be `stream.EOS`  to conform to `stream.Iterator`

Reply:
I think we want some sort of utility function to call a callback for each valid row in a Stream.

Reply:
`stream.ForEach`?

Reply:
Yup.  Thinking about this more, I think we need a way of converting sqlx.Rows into a Stream, and a way of converting an Stream of []sqlx.Rows into a Stream.  This will let us always handle SQL queries with our iterator API, and let us represent the likely-to-be-very-common "paged" approach in the same way; the consumer doesn't have to care which implementation we chose.

Reply:
That's a good idea, but I don't think that should block this PR.

Reply:
Let's move that conversation here.
https://github.com/pachyderm/pachyderm/pull/8959

If it's usefui, then there will be motivation to use it for the Commit and Repo iterators, and we can migrate the Projects Iterator then.